src/clr-angular/data/datagrid/render/main-renderer.ts

```diff
@@ -189,4 +194,16 @@ export class DatagridMainRenderer<T = any> implements AfterContentInit, AfterVie
       this.columnsSizesStable = true;
     }
   }
+
+  private setHeaderOrders(): void {
+    this.headers.forEach((header, index) => {
+      // set initial flex order
+      header.setFlexOrder(index);
+    });
+
+    // set orders array with headers ColumnOrder
+    this.columnOrderCoordinatorService.orderModels = this.headers.map((header, index) => {
```

Comment:
You're only setting `ColumnOrdersCoordinatorService.orderModels` on initialization. This breaks for dynamic columns.

Reply:
Yes, I have implemented a method that responds and addresses dynamic columns in the next upcoming PR, which also includes changes that actually reorders columns.